alphalens/performance.py

```diff
@@ -432,20 +436,82 @@ def average_cumulative_return_by_quantile(quantized_factor,
         How many periods after factor to plot
     demeaned : bool, optional
         Compute demeaned mean returns (long short portfolio)
+    group_adjust : bool
+        Returns demeaning will occur on the group level (group
+        neutral portfolio)
+    by_group : bool
+        If True, compute cumulative returns separately for each group
     Returns
     -------
-    pd.DataFrame indexed by quantile (level 0) and mean/std
-    (level 1) and the values on the columns in range from
-    -periods_before to periods_after
+    cumulative returns and std deviation : pd.DataFrame
+        A MultiIndex DataFrame indexed by quantile (level 0) and mean/std
+        (level 1) and the values on the columns in range from
+        -periods_before to periods_after
+        If by_group=True the index will have an additional 'group' level
+        ::
+            ---------------------------------------------------
+                        |       | -2  | -1  |  0  |  1  | ...
+            ---------------------------------------------------
+              quantile  |       |     |     |     |     |
+            ---------------------------------------------------
+                        | mean  |  x  |  x  |  x  |  x  |
+                 1      ---------------------------------------
+                        | std   |  x  |  x  |  x  |  x  |
+            ---------------------------------------------------
+                        | mean  |  x  |  x  |  x  |  x  |
+                 2      ---------------------------------------
+                        | std   |  x  |  x  |  x  |  x  |
+            ---------------------------------------------------
+                ...     |                 ...
+            ---------------------------------------------------
     """
 
-    def average_cumulative_return(q_fact):
-        demean = quantized_factor if demeaned else None
-        q_returns = utils.common_start_returns(q_fact, prices,
-                                               periods_before, periods_after,
-                                               True, True, demean)
+    def cumulative_return(q_fact, demean_by):
+        return utils.common_start_returns(q_fact, prices,
+                                          periods_before,
+                                          periods_after,
+                                          True, True, demean_by)
+
+    def average_cumulative_return(q_fact, demean_by):
+        q_returns = cumulative_return(q_fact, demean_by)
         return pd.DataFrame({'mean': q_returns.mean(axis=1),
                              'std': q_returns.std(axis=1)}).T
 
-    return quantized_factor.groupby(quantized_factor) \
-        .apply(average_cumulative_return)
+    if by_group:
```

Comment:
This code could use some comments.

Reply:
I usually comment a lots but because Alphalens don't have comments at all I have almost always refrained from adding them, I thought it was kind of a policy :D

Reply:
Haha, not at all :)